### PR TITLE
Add LINKsystem, remove dependency on Target from the parser

### PR DESCRIPTION
**Pull request by ibuclaw**
As requested by @yebblies - I don't think this should leak to the glue, if it does, then that should be plugged.

### CODE REVIEW

File: src/ddmd/attrib.d [hunk 1]
```diff
@@ -499,7 +500,7 @@ extern (C++) final class LinkDeclaration : AttribDeclaration
     {
         super(decl);
         //printf("LinkDeclaration(linkage = %d, decl = %p)\n", p, decl);
-        linkage = p;
+        linkage = (p != LINKsystem) ? p : Target.systemLinkage();
```
Comment (WalterBright) [inline src/ddmd/attrib.d]: This would flow better as:
```
(p == LINKsystem) ? Target.systemLinkage() : p;
```

Comment (ibuclaw) [inline src/ddmd/attrib.d]: Yeah, I did, but then I remembered some obscure optimization detail that dmd considers the first branch as hot.  Maybe that's one micro-optimization too far though. ;-)

**Comment by ibuclaw** [comment]
Rebased.

**Comment by yebblies** [comment]
I don't like that we're discarding information as we build the parse tree, but it is the same with the current version.

**Comment by ibuclaw** [comment]
> I don't like that we're discarding information as we build the parse tree

Mehhh... At least two errors have improved.

If I were thinking about having the frontend as a library, you would have a point though.

**Comment by ibuclaw** [comment]
Is someone going to fix Jenkins, or...

**Pull request merged**